src/mono/wasm/templates/templates/console/app-support.cjs

```diff
@@ -0,0 +1,427 @@
+            diagnostic_tracing = true;
+        } else if (currentArg.startsWith("--working-dir=")) {
+            const arg = currentArg.substring("--working-dir=".length);
+            working_dir = arg;
+        } else if (currentArg == "--debug") {
+            is_debugging = true;
+        } else if (currentArg == "--no-forward-console") {
+            forward_console = false;
+        } else {
+            break;
+        }
+        incomingArguments = incomingArguments.slice(1);
+    }
+
+    // cheap way to let the testing infrastructure know we're running in a browser context (or not)
+    setenv["IsBrowserDomSupported"] = is_browser.toString().toLowerCase();
+    setenv["IsNodeJS"] = is_node.toString().toLowerCase();
+
+    console.log("Application arguments: " + incomingArguments.join(' '));
+
+    return {
+        applicationArgs: incomingArguments,
+        profilers,
+        setenv,
+        runtime_args,
+        enable_gc,
+        diagnostic_tracing,
+        working_dir,
+    }
+}
+
+function proxyConsoleMethod(prefix, func, asJson) {
+    return function () {
+        try {
+            const args = [...arguments];
+            let payload = args[0];
+            if (payload === undefined) payload = 'undefined';
+            else if (payload === null) payload = 'null';
+            else if (typeof payload === 'function') payload = payload.toString();
+            else if (typeof payload !== 'string') {
+                try {
+                    payload = JSON.stringify(payload);
+                } catch (e) {
+                    payload = payload.toString();
+                }
+            }
+
+            if (payload.startsWith("STARTRESULTXML")) {
+                originalConsole.log('Sending RESULTXML')
+                func(payload);
+            }
+            else if (asJson) {
+                func(JSON.stringify({
+                    method: prefix,
+                    payload: payload,
+                    arguments: args
+                }));
+            } else {
+                func([prefix + payload, ...args.slice(1)]);
+            }
+        } catch (err) {
+            originalConsole.error(`proxyConsole failed: ${err}`)
+        }
+    };
+};
+
+// this can't be function because of `arguments` scope
+try {
+    if (is_node) {
+        processedArguments = processArguments(process.argv.slice(2));
+    } else if (is_browser) {
+        // We expect to be run by tests/runtime/run.js which passes in the arguments using http parameters
+        const url = new URL(decodeURI(window.location));
+        let urlArguments = []
+        for (let param of url.searchParams) {
+            if (param[0] == "arg") {
+                urlArguments.push(param[1]);
+            }
+        }
+        processedArguments = processArguments(urlArguments);
+    } else if (typeof arguments !== "undefined") {
+        processedArguments = processArguments(Array.from(arguments));
```

Comment:
this is for v8

Reply:
`processArguments` could be lifted to `dotnet.js` and added to it's `DotnetModuleConfig`